frontend/src/components/common/MinimalHeader.tsx

```diff
@@ -0,0 +1,35 @@
+import LogoTextGroup from "./LogoTextGroup"
+import { UserButton } from "@clerk/clerk-react"
+
+/**
+ * MinimalHeader Component
+ *
+ * A simplified header component that includes only the logo and user button.
+ *
+ * @returns {JSX.Element} The minimal header.
+ */
+const MinimalHeader = () => {
```

Comment:
```suggestion
const MinimalHeader: React.FC = () => {
```

because we have this doc @returns {JSX.Element} The minimal header.  
so better add change to prevent IDE error